### DISCUSSION TITLE
Docker volume for assets?

### DISCUSSION THREAD

**Discussion by AuspeXeu** [requarks/wiki#6564] [Error / Bug Report] [answered]
In your docker-compose examples I cannot find a volume mount to ensure that assets survive container upgrades?

**Comment by fnafox** [comment]
afaik assets are saved in the database, filesystem is only used as cache.

**Reply by AuspeXeu** [reply]
Removing the images from the assets folder (`/wiki/data/repo`) to another location causes all images inside the wiki to 404.

**Reply by P4mes4n** [reply]
Strange, I dont have a `/wiki/data/repo` in my docker instance.
Did you try the purge cache options in the wiki, to see if the images are fetched again?

**Reply by AuspeXeu** [reply]
Tried all of this and just had another image gone missing which I uploaded this morning. It's really rather frustrating.

**Comment by NGPixel** [comment, marked as answer]
Assets are saved in the DB. You can destroy and recreate the container without worrying about assets.

The path you mention (`/wiki/data/repo`) is for git storage synchronization, which you have enabled. Manually deleting files from there will obviously delete them in Wiki.js, because of the sync.

**Reply by AuspeXeu** [reply]
This means if the container gets destroyed and recreated, it will pull it from git?

**Reply by NGPixel** [reply]
Correct.

**Comment by jwilleke** [comment]
Can someone provide the docker entries required for using external storage (or sync) of pages?
